python/cudf/dataframe/index.py

```diff
@@ -588,16 +636,12 @@ def as_index(arbitrary, name=None):
     # This function should probably be moved to Index.__new__
     if isinstance(arbitrary, Index):
         return arbitrary
-    elif isinstance(arbitrary, NumericalColumn):
+    elif isinstance(arbitrary, (StringColumn, NumericalColumn)):
```

Comment:
If it's a `StringColumn` you'd want to return a `StringIndex`, no?

Reply:
Yeah probably. :) I think this was a Chicken/Egg item, fixing one issue before I created the full StringIndex perhaps. Should be an easy fix.

Reply:
`StringColumn.copy()` shouldn't do anything related to `as_index`. Digging in now to see if anything pops up.

Reply:
After merging in the latest `branch-0.7` it, uh, went away.

Reply:
Corrected as proposed.